File: Java/0268_Missing_Number.java
// 2020-08-07 17:15:07

class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length, ret = n;
        for (int i = 0; i < n; i++) {
            ret = ret ^ i ^ nums[i];
        }
        return ret;
    }
}

File: C++/0268_missingnumber.cc
// 2020-08-08 16:24:48

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int ret = nums.size();
        for (int i = 0; i < nums.size(); i++) {
            ret = ret ^ nums[i] ^ i;
        }
        return ret;
    }
};